src/fund.rs

```diff
@@ -0,0 +1,93 @@
+use ethcore::{
+    self,
+    state::{CleanupMode, State},
+};
+use ethereum_types::{Address, H256, U256};
+use oasis_ethwasi_runtime_common::parity::NullBackend;
+extern crate chrono;
+use chrono::{DateTime, Datelike, NaiveDateTime, Utc};
+extern crate date_time;
+use byteorder::{BigEndian, ReadBytesExt};
+use date_time::month_tuple::MonthTuple;
+use std::{convert::TryInto, str::FromStr};
+
+fn timestamp_to_mt(timestamp: i64) -> MonthTuple {
+    let dt = DateTime::<Utc>::from_utc(NaiveDateTime::from_timestamp(timestamp, 0), Utc);
+    return MonthTuple::new(
+        dt.year().try_into().unwrap(),
+        dt.month().try_into().unwrap(),
+    )
+    .unwrap();
+}
+
+pub struct FundManager;
+
+impl FundManager {
+    /// Unlock address.
+    const BENEFICIARY: &'static str = "7110316b618d20d0c44728ac2a3d683536ea682b";
+    /// Adjust period.
+    const PERIOD: u32 = 20;
+    // Factor
+    const FACTOR: u32 = 2;
+    /// Total Amount is 21000000000000000000000000 wei.
+    const TOTAL_AMOUNT: &'static str = "115EEC47F6CF7E35000000";
+    /// Initial unlock bucket is 525000000000000000000000 wei.
+    const INITIAL_BUCKET: &'static str = "6F2C4E995EC98E200000";
+
+    // There is a total fixed supply of 21 million OETHs.
+    // The blockchain unlocks 525,000 OETHs every month in the first 20 months and
+    // the monthly release is cut to 1/2 every 20 months.
+
+    pub fn try_unlock(timestamp: u64, state: &mut State<NullBackend>) {
+        let beneficiary = Address::from_str(FundManager::BENEFICIARY).unwrap();
+        let total_amount = U256::from_str(FundManager::TOTAL_AMOUNT).unwrap();
+        let initial_bucket = U256::from_str(FundManager::INITIAL_BUCKET).unwrap();
+
+        // +---------------+----------------+-----------+-----------+
+        // | Storage field |       0:23     |   24:27   |   28:32   |
+        // +---------------+----------------+-----------+-----------+
+        // |             0 |     reserved   |    init timestamp     |
+        // +---------------+----------+-----------------+-----------+
+        // |             1 |            reserved        |   round   |
+        // +---------------+----------+-----------------+-----------+
+        // |             2 |              unlock amount             |
+        // +---------------+----------+-----------------+-----------+
+        let value = state.storage_at(&beneficiary, &H256::from(0)).unwrap();
+        let init_timestamp = value.get(24..).unwrap().read_i64::<BigEndian>().unwrap();
+        let value = state.storage_at(&beneficiary, &H256::from(1)).unwrap();
+        let mut unlock_round = value.get(28..).unwrap().read_u32::<BigEndian>().unwrap();
+        let value = state.storage_at(&beneficiary, &H256::from(2)).unwrap();
+        let mut unlock_amount: U256 = value.into();
+
+        if unlock_amount < total_amount {
+            let mut last_ym = timestamp_to_mt(init_timestamp as i64);
```

Comment:
You variable name should have more comments.

Reply:
done.